protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java

```diff
@@ -0,0 +1,471 @@
+package org.geysermc.mcprotocollib.auth;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.geysermc.mcprotocollib.auth.util.UndashedUUIDAdapter;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Information about a user profile.
+ */
+public class GameProfile {
+    private static final String[] WHITELISTED_DOMAINS = {".minecraft.net", ".mojang.com"};
```

Comment:
Should we not include the block list like the mojang authlib?

Currently blocks:
- bugs.mojang.com
- education.minecraft.net
- feedback.minecraft.net

Seen in `com.mojang.authlib.yggdrasil.TextureUrlChecker`